comp/trace/config/setup.go

```diff
@@ -223,6 +224,22 @@ func applyDatadogConfig(c *config.AgentConfig, core corecompcfg.Component) error
 	}
 	c.PeerServiceAggregation = core.GetBool("apm_config.peer_service_aggregation")
 	c.ComputeStatsBySpanKind = core.GetBool("apm_config.compute_stats_by_span_kind")
+	if core.IsSet("apm_config.custom_tags") {
+		// TODO: make func, test it separately
```

Comment:
TODO: add defaults here

If the field is set, then we can consider that to be opt-in.